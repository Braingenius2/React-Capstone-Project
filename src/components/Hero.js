import React from 'react';
import PropTypes from 'prop-types';

const Hero = ({ image, name, details }) => (
  <div className="hero flex items-center justify-center bg-pink-500">
    <div className="w-1/2">
      <img src={image} alt={name} className="w-full max-h-200" />
    </div>
    <div className="ml-4 w-1/3">
      <h2 className="text-2xl font-bold ">{ name }</h2>
      {
        details.map(
          ({ stats, title }) => (
            <p key={title} className="text-lg">
              { stats }
              {' '}
              { title }
            </p>
          ),
        )
      }
    </div>
  </div>
);

Hero.defaultProps = {
  image: '',
  name: '',
  details: [],
};

Hero.propTypes = {
  image: PropTypes.string,
  name: PropTypes.string,
  details: PropTypes.arrayOf(PropTypes.shape({
    stats: PropTypes.string,
    text: PropTypes.string,
  })),
};

export default Hero;
