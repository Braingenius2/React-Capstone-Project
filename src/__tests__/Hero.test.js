import React from 'react';
import { render } from '@testing-library/react';
import Hero from '../components/Hero';

test('renders Hero component', () => {
  const details = [
    { stats: '1000', text: 'cases' },
    { stats: '500', text: 'recovered' },
  ];
  const { getByText } = render(<Hero name="Continent" details={details} />);
  const casesElement = getByText('1000 cases');
  const recoveredElement = getByText('500 recovered');
  expect(casesElement).toBeInTheDocument();
  expect(recoveredElement).toBeInTheDocument();
});
