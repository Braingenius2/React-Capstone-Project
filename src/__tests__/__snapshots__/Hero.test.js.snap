// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Hero component should render correctly into the DOM 1`] = `
<div>
  <div
    class="hero flex items-center justify-center bg-pink-500"
    style="min-height: 70vh;"
  >
    <div
      class="w-1/2"
    >
      <img
        alt="Asia"
        class="w-full max-h-96"
        src=""
      />
    </div>
    <div
      class="ml-4 w-1/3"
    >
      <h2
        class="text-3xl font-bold "
      >
        Asia
      </h2>
    </div>
  </div>
</div>
`;
